webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -143,14 +145,26 @@ class Offer extends Component {
   </div>
       )}
 
+    componentWillMount(){
+      userService.get().then(result=> {
+        this.setState({
+            user:{
+              user_id:result.user_id
+            }
+        });
+        DEFAULT_USER_ID = result.user_id;
+      });
+
+    }
+
     componentDidMount() {
           this.getOffer();
           this.setState({
             user: JSON.parse(localStorage.getItem("user")),
             loading: false,
             buttonLoading: false
           });
-          profileService.getUserProfile(user.user_id)
+          profileService.getUserProfile(DEFAULT_USER_ID)
```

Comment:
Do not use this detault user ID. It makes no sense.  
You need to user the one in local storage that you set above.
surely it's `state.user.id`?  
If it ends up being null then chances are you have a async issue.